test/gtest/uct/ib/test_dc.cc

```diff
@@ -658,6 +650,47 @@ UCS_TEST_P(test_dc_flow_control, dci_leak)
 
 UCT_DC_INSTANTIATE_TEST_CASE(test_dc_flow_control)
 
+
+class test_dc_flow_control_with_hard_req_resend : public test_dc_flow_control
+{
+protected:
+    virtual void init() {
+        ucs_status_t status = uct_config_modify(m_iface_config,
+                                                "DC_FC_HARD_REQ_TIMEOUT",
+                                                "0.1s");
+        if (status != UCS_OK) {
+            UCS_TEST_ABORT("cannot set DC_FC_HARD_REQ_TIMEOUT");
+        }
+        test_dc_flow_control::init();
+    }
+
+    virtual void wait_fc_hard_resend(entity *e)
+    {
+        uct_dc_mlx5_iface_t *iface = ucs_derived_of(e->iface(),
+                                                    uct_dc_mlx5_iface_t);
+        uct_dc_mlx5_ep_t *ep       = ucs_derived_of(e->ep(0),
+                                                    uct_dc_mlx5_ep_t);
+
+        khiter_t it = kh_get(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash,
+                             (uint64_t)ep);
+        ASSERT_NE(it, kh_end(&iface->tx.fc_hash));
+
+        uct_dc_mlx5_ep_fc_entry_t *fc_entry = &kh_value(&iface->tx.fc_hash,
+                                                        it);
+        ucs_time_t fc_send_time = fc_entry->send_time;
+        wait_for_value_change(&fc_entry->send_time, e);
```

Comment:
do we need to avoid progress on target side?

Reply:
no, we don't need to progress a target side, otherwise - it will send FC_PURE_GRANT which is not needed